jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/AttachHandler.java

```diff
@@ -323,6 +332,88 @@ private boolean initialize() throws IOException {
 		syncFileLock = new FileLock(syncFileObjectTemp.getAbsolutePath(), TargetDirectory.SYNC_FILE_PERMISSIONS);
 		/* give other users read permission to the sync file */
 
+		if (IPC.isLinux) {
+			// Update AttachAPI control file access time every sleepDays (8 days by default).
+			// This is to prevent Linux systemd from deleting the ageing files within /tmp after 10 days.
+			// More details at https://github.com/eclipse-openj9/openj9/issues/18720
+			int sleepDays = 8;
+			String fileAccessUpdateTime = internalProperties.getProperty("com.ibm.tools.attach.fileAccessUpdateTime", "8");
+			IPC.logMessage("fileAccessUpdateTime = " + fileAccessUpdateTime); //$NON-NLS-1$
+			try {
+				sleepDays = new Integer(fileAccessUpdateTime).intValue();  //$NON-NLS-1$
+			} catch (NumberFormatException nfe) {
+				// ignore this non-fatal exception, and use the default value
+			}
+			IPC.logMessage("fileAccessTimeUpdaterThread sleepDays  = " + sleepDays); //$NON-NLS-1$
+			if (sleepDays > 0) {
+				final long targetIntervalMillis = DAYS.toMillis(sleepDays);
+				final String lastAccessTime = "lastAccessTime"; //$NON-NLS-1$
+				final File commonDir = CommonDirectory.getCommonDirFileObject();
+				fileAccessTimeUpdaterThread = new Thread("Attach API update file access time") { //$NON-NLS-1$
+					public void run() {
+						VM.markCurrentThreadAsSystem();
+
+						long lastMillis = System.currentTimeMillis();
+						long sleepMillis = targetIntervalMillis;
+						for (;;) {
+							try {
+								Thread.sleep(sleepMillis);
+							} catch (InterruptedException ie) {
+								// ignore this non-fatal exception
+								IPC.logMessage("fileAccessTimeUpdaterThread received an InterruptedException : " + ie.getMessage()); //$NON-NLS-1$
+							}
+							if (isAttachApiTerminated()) {
+								IPC.logMessage("AttachAPI was terminated, fileAccessTimeUpdaterThread exits"); //$NON-NLS-1$
+								break;
+							}
+							long currentMillis = System.currentTimeMillis();
+							long passedMillis = currentMillis - lastMillis;
+							if (passedMillis >= sleepMillis) {
+								FileTime fileTime = FileTime.fromMillis(currentMillis);
+								IPC.logMessage("fileAccessTimeUpdaterThread currentMillis = " + currentMillis); //$NON-NLS-1$
+								try {
+									// Common directory : _attachlock
+									File attachLock = new File(commonDir, CommonDirectory.ATTACH_LOCK);
+									if (attachLock.exists()) {
+										// _attachlock only appears when there was an attaching request
+										Files.setAttribute(Paths.get(attachLock.getPath()), lastAccessTime, fileTime);
+										IPC.logMessage("fileAccessTimeUpdaterThread updated access time = " + attachLock); //$NON-NLS-1$
+									}
+									// Common directory : _controller
+									Files.setAttribute(Paths.get(new File(commonDir, CommonDirectory.CONTROLLER_LOCKFILE).getPath()),
```

Comment:
This should use `Path.get(String first, String... more)`
```java
Paths.get(commonDir.getPath(), CommonDirectory.CONTROLLER_LOCKFILE)
```
(there's no need to create an intermediate `File` object)

Reply:
Removed the `intermediate` File object.